rudder-agent/SPECS/rudder-agent.spec

```diff
@@ -301,8 +301,7 @@ TRY_CFLAGS="-fPIE -fstack-protector"
 
 FLAG_TEST_FILE=`mktemp /tmp/hardening.XXXXXX`
 echo "void main() {}" > "${FLAG_TEST_FILE}.c"
-gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" 2>/dev/null
-if [ $? -eq 0 ]
+if [ $(gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" >/dev/null 2>&1; echo $?) -eq 0 ]
```

Comment:
It does not fail in set -e 

I tried with the following script: 

     #!/bin/sh

      set -e

    SECURE_CFLAGS="NO FLAGS"
    TRY_LDFLAGS="-pie -Wl,-z,relro -Wl,-z,now"
    TRY_CFLAGS="-fPIE -fstack-protector"
    FLAG_TEST_FILE=`mktemp /tmp/hardening.XXXXXX`
    echo "void main() {}" > "${FLAG_TEST_FILE}.c"
    if gcc ${TRY_CFLAGS} ${TRY_LDFLAGS} -o "${FLAG_TEST_FILE}" "${FLAG_TEST_FILE}.c" >/dev/null 2>&1
     then
       SECURE_CFLAGS="${TRY_CFLAGS}"
       SECURE_LDFLAGS="${TRY_LDFLAGS}"
    fi
    echo $SECURE_CFLAGS